test/Array/array_sort_random.js

```diff
@@ -0,0 +1,160 @@
+                    "s", "t", "u", "v", "w", "x", "y", "z"];
+
+function getRandomObjArray(size)
+{
+    const arr = [];
+    for (let i = 0; i < size; ++i)
+    {
+        arr[i] = {
+            value : Math.random() * 100 | 0,
+            index : i
+        }
+    }
+    return arr;
+}
+
+function getRandomStringArray(size)
+{
+    const arr = [];
+    for (let i = 0; i < size; ++i)
+    {
+        let value = "";
+        const len = Math.random() * 5 | 0;
+        for (let j = 0; j < len; ++j)
+        {
+            value += alphabet[Math.random() * 26 |0];
+        }
+        arr[i] = value;
+    }
+    return arr;
+}
+
+function getRandomIntArray(size)
+{
+    const arr = [];
+    for (let i = 0; i < size; ++i)
+    {
+        arr[i] = Math.random() * 100 | 0;
+    }
+    return arr;
+}
+
+function testRandomIntSort(size)
+{
+    const unsorted = getRandomIntArray(size);
+
+    // Copy the array and sort it
+    const sorted = unsorted.slice();
+    sorted.sort(function (a, b){ return 0;});
+
+    for (let i = 0; i < size; ++i)
+    {
+        if (sorted[i] !== unsorted[i])
+        {
+            throw new Error ("Sort not stable at index " + i);
+        }
+    }
+
+    sorted.sort();
+
+    for (let i = 1; i < size; ++i)
+    {
+        if ("" + sorted[i-1] > "" + sorted[i])
+        {
+            print (sorted);
+            throw new Error ("Default sort has not sorted by strings at " + i);
+        }
+    }
+
+    sorted.sort (function (a, b){ return a - b;});
+
+    // Verify that the array is sorted
+    for (let i = 1; i < size; ++i)
+    {
+        // Sort has not completed correctly
+        if (sorted[i] < sorted[i - 1])
+        {
+            print (`Unsorted: ${unsorted}`);
+            print (`Sorted: ${sorted}`);
+            throw new Error(`Array is not sorted correctly at index '${i}'`);
+        }
+    }
+}
+
+
+function testRandomStringSort(size)
+{
+    const unsorted = getRandomStringArray(size);
+
+    // Copy the array and sort it
+    const sorted = unsorted.slice();
+    sorted.sort(function (a, b){ return 0;});
+
+    for (let i = 0; i < size; ++i)
+    {
+        if (sorted[i] !== unsorted[i])
+        {
+            throw new Error ("Sort not stable at index " + i);
+        }
+    }
+
+    sorted.sort();
+
+    for (let i = 1; i < size; ++i)
+    {
+        if (sorted[i-1] > sorted[i])
+        {
+            throw new Error ("Default sort has not sorted by strings");
+        }
+    }
+}
+
+function testRandomObjSort(size)
+{
+    const array = getRandomObjArray(size);
+
+    // stable default sort on objects without custom toString should not re-order
+    array.sort();
+    for (let i = 0; i < size; ++i)
+    {
+        if (array[i].index !== i)
+        {
+            throw new Error ("Sort not stable at index " + i);
+        }
+    }
+
+    array.sort(function (a, b) { return a.value - b.value; });
+
+    for (let i = 1; i < size; ++i)
+    {
+        if (array[i-1].value > array[i].value)
```

Comment:
Also check that if `array[i-1] === array[i]`, then `array[i-1].index < array[i].index` (to verify that stability is maintained while moving things around).